vshard/util.lua

```diff
@@ -279,6 +279,32 @@ local function fiber_cond_wait(cond, timeout)
     return nil, lerror.make(err)
 end
 
+local function future_wait_xc(cond, timeout)
+    -- Handle negative timeouts in order to avoid usage error.
+    -- The idea is the same as in fiber_cond_wait_xc function.
+    if timeout and timeout < 0 then
+        error(lerror.timeout())
+    end
+    -- pcall() is inevitable as wait_result can throw an error
+    -- if executed in net.box's trigger for example.
+    local res, err = cond:wait_result(timeout)
```

Comment:
The comment should be where you do `pcall()`. Here is looks strange that you mention `pcall()` and don't do it.